bin/varnishd/cache/cache_esi_deliver.c

```diff
@@ -899,4 +904,9 @@ ved_deliver(struct req *req, struct boc *boc, int wantbody)
 		req->doclose = SC_REM_CLOSE;
 
 	req->acct.resp_bodybytes += VDP_Close(req->vdc);
+
+	if (i && ecx->abrt) {
+		req->top->topreq->vdc->retval = -1;
+		req->top->topreq->doclose = req->doclose;
```

Comment:
This looks deeply suspect to me.  Doesn't the VDP stack not already ensure this ?

Reply:
See test case, apparently not.